src/Microsoft.SqlTools.ServiceLayer/LanguageServices/ConnectedBindingQueue.cs

```diff
@@ -81,20 +81,7 @@ public virtual string AddConnectionContext(ConnectionInfo connInfo, bool overwri
                 try
                 {
                     bindingContext.BindingLock.Reset();
-
-                    // increase the connection timeout to at least 30 seconds and and build connection string
-                    // enable PersistSecurityInfo to handle issues in SMO where the connection context is lost in reconnections
-                    int? originalTimeout = connInfo.ConnectionDetails.ConnectTimeout;
-                    bool? originalPersistSecurityInfo = connInfo.ConnectionDetails.PersistSecurityInfo;
-                    connInfo.ConnectionDetails.ConnectTimeout = Math.Max(DefaultMinimumConnectionTimeout, originalTimeout ?? 0);
-                    connInfo.ConnectionDetails.PersistSecurityInfo = true;
-                    string connectionString = ConnectionService.BuildConnectionString(connInfo.ConnectionDetails);
-                    connInfo.ConnectionDetails.ConnectTimeout = originalTimeout;
-                    connInfo.ConnectionDetails.PersistSecurityInfo = originalPersistSecurityInfo;
-
-                    // open a dedicated binding server connection
-                    SqlConnection sqlConn = new SqlConnection(connectionString);                    
-                    sqlConn.Open();
+                    SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo);
```

Comment:
>sqlConn [](start = 34, length = 7)

sqlConn can be null so we should check for null here

Reply:
it looks like throwing here is the correct thing to do so that we set IsConnected to false in the catch block.  If we null check we'd need another clause to set IsConnected=false...basically we just changed a SqlException to a NullRefException that gets handled in the same place (and added better logging in ConnectionService for the exception).